File: src/Oisys.Web/Models/Customer.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace OisysNew.Models
{
    /// <summary>
    /// <see cref="Customer"/> class Customer object.
    /// </summary>
    public class Customer : SoftDeletableModel
    {
        /// <summary>
        /// Gets or sets property Name.
        /// </summary>
        public string Name { get; set; }

        /// <summary>
        /// Gets or sets property Phone Number.
        /// </summary>
        public string ContactNumber { get; set; }

        /// <summary>
        /// Gets or sets property Contact Person.
        /// </summary>
        public string ContactPerson { get; set; }

        /// <summary>
        /// Gets or sets property Email.
        /// </summary>
        public string Email { get; set; }

        /// <summary>
        /// Gets or sets property Address.
        /// </summary>
        public string Address { get; set; }

        /// <summary>
        /// Gets or sets property City.
        /// </summary>
        public long CityId { get; set; }

        /// <summary>
        /// Gets or sets property Province.
        /// </summary>
        public long ProvinceId { get; set; }

        /// <summary>
        /// Gets or sets property Terms.
        /// </summary>
        public string Terms { get; set; }

        /// <summary>
        /// Gets or sets property Discount.
        /// </summary>
        public decimal Discount { get; set; }

        /// <summary>
        /// Gets or sets property Price List.
        /// </summary>
        public PriceList PriceList { get; set; }

        /// <summary>
        /// Gets or sets property Keywords.
        /// </summary>
        public string Keywords { get; set; }

        /// <summary>
        /// Gets or sets property City.
        /// </summary>
        public virtual City City { get; set; }

        /// <summary>
        /// Gets or sets property Province.
        /// </summary>
        public virtual Province Province { get; set; }

        /// <summary>
        /// Gets or sets property Customer transactions.
        /// </summary>
        public virtual ICollection<CustomerTransaction> Transactions { get; set; }
    }

    public enum PriceList
    {
        [Display(Name = "Main Price")]
        MainPrice = 1,
        [Display(Name = "Walk-In Price")]
        WalkInPrice = 2,
        [Display(Name = "N.E. Price")]
        NEPrice = 3
    }
}

File: src/Oisys.Web/ClientApp/src/app/shared/models/customer.ts
import { PriceList } from "./price-list";
import { Province } from "./Province";
import { City } from "./city";

export class Customer {
  public id: number;
  public name: string;
  public email: string;
  public contactNumber: string;
  public contactPerson: string;
  public address: string;
  public cityId: number;
  public provinceId: number;
  public terms: string;
  public discount: number;
  public priceListId: number;
  public provinceName: string;
  public cityName: string;

  private _selectedProvince: Province;
  get selectedProvince() {
    return this._selectedProvince;
  }
  set selectedProvince(prov: Province) {
    if (prov) {
      this._selectedProvince = prov;
      this.provinceId = prov.id;
    }
  }

  private _selectedCity: City;
  get selectedCity() {
    return this._selectedCity;
  }
  set selectedCity(city: City) {
    if (city) {
      this._selectedCity = city;
      this.cityId = city.id;
    } else {
      this._selectedCity = undefined;
      this.cityId = 0;
    }
  }

  constructor();
  constructor(customer: Customer);
  constructor(customer?: any) {
    this.id = customer && customer.id || 0;
    this.name = customer && customer.name || '';
    this.email = customer && customer.email || '';
    this.contactNumber = customer && customer.contactNumber || '';
    this.contactPerson = customer && customer.contactPerson || '';
    this.address = customer && customer.address || '';
    this.terms = customer && customer.terms || '';
    this.discount = customer && customer.discount || '';

    this.provinceId = customer && customer.provinceId || 0;
    this.provinceName = customer && customer.provinceName || '';

    this.cityId = customer && customer.cityId || 0;
    this.cityName = customer && customer.cityName || '';

    this.priceListId = customer && customer.priceListId || PriceList["Main Price"];
  }
}
